### DISCUSSION TITLE
[Discussion] New limits in nuget.org's Search API

### DISCUSSION THREAD

**Discussion by jcjiang** [NuGet/Home#10945] [General]
Discussion issue for **https://github.com/NuGet/Announcements/issues/59: New limits in nuget.org's Search API**

**Comment by BahJiy** [comment]
With the limitation on v3 search API, how would I query for all the packages from an owner/author?

Right now, I'm trying to clone the all C# NuGet packages from **Microsoft**:

`https://azuresearch-usnc.nuget.org/query?q=owner:Microsoft&skip=0&take=1000`

However, there are more than 4,000 packages (the theoretically max returnable by SearchQueryService).

My current workaround is to filter by namespace (i.e. System, Microsoft.*) or target shared owners (i.e. dotnetframework, entityframework); however, there's no guarantee it captures all packages. 

Iterating through the catalog is slow since it doesn't directly provide owner information; I would have to go through all the **id** since the beginning of NuGet, then verify the owners while building an internal database of all unique packages to download.
